cloud_experts_tutorials/cloud-experts-using-aws-ack.adoc

```diff
@@ -0,0 +1,242 @@
+//title: Using AWS Controllers for Kubernetes (ACK) on ROSA
+//weight: 1
+//tags: ["AWS", "ROSA"]
+//authors:
+//  - Paul Czarkowski
+//  - Connor Wooley
+//---
+
+link:https://aws-controllers-k8s.github.io/community/[AWS Controllers for Kubernetes] (ACK) lets you define and use AWS service resources directly from {product-title} (ROSA). With ACK, you can take advantage of AWS-managed services for your applications without needing to define resources outside of the cluster or run services that provide supporting capabilities like databases or message queues within the cluster.
+
+Users can install various ACK operators directly from OperatorHub. This makes it relatively easy to get started and start using it with your applications. This controller is a component of the AWS Controller for Kubernetes project. This project is currently in developer preview.
+
+This tutorial shows you how to use the ACK S3 operator as an example, but can be adapted for any other ACK operator in the OperatorHub of your cluster.
+
+== Prerequisites
+
+* A ROSA cluster
+* A user account with `cluster-admin` privileges
+* You have access to the OpenShift CLI (`oc`).
+* You have access to the AWS CLI (`aws`).
+
+=== Environment Setup
+
+. Configure the following environment variables:
++
+[source,terminal]
+----
+$ export CLUSTER_NAME=$(oc get infrastructure cluster -o=jsonpath="{.status.infrastructureName}"  | sed 's/-[a-z0-9]\{5\}$//')
+$ export REGION=$(rosa describe cluster -c ${ROSA_CLUSTER_NAME} --output json | jq -r .region.id)
+$ export OIDC_ENDPOINT=$(oc get authentication.config.openshift.io cluster -o json | jq -r .spec.serviceAccountIssuer | sed  's|^https://||')
+$ export AWS_ACCOUNT_ID=`aws sts get-caller-identity --query Account --output text`
+$ export ACK_SERVICE=s3
+$ export ACK_SERVICE_ACCOUNT=ack-${ACK_SERVICE}-controller
+$ export POLICY_ARN=arn:aws:iam::aws:policy/AmazonS3FullAccess
+$ export AWS_PAGER=""
+$ export SCRATCH="/tmp/${ROSA_CLUSTER_NAME}/ack"
+$ mkdir -p ${SCRATCH}
+$ echo "Cluster: ${ROSA_CLUSTER_NAME}, Region: ${REGION}, OIDC Endpoint: ${OIDC_ENDPOINT}, AWS Account ID: ${AWS_ACCOUNT_ID}"
+----
+
+== Prepare AWS Account
+
+. Create an AWS IAM trust policy for the ACK operator:
++
+[source,terminal]
+----
+$ cat <<EOF > "${SCRATCH}/trust-policy.json"
+{
+ "Version": "2012-10-17",
+ "Statement": [
+ {
+ "Effect": "Allow",
+ "Condition": {
+   "StringEquals" : {
+     "${OIDC_ENDPOINT}:sub": "system:serviceaccount:ack-system:${ACK_SERVICE_ACCOUNT}"
+   }
+ },
+ "Principal": {
+   "Federated": "arn:aws:iam::$AWS_ACCOUNT_ID:oidc-provider/${OIDC_ENDPOINT}"
+ },
+ "Action": "sts:AssumeRoleWithWebIdentity"
+ }
+ ]
+}
+EOF
+----
++
+. Create an AWS IAM role for the ACK operator to assume with the `AmazonS3FullAccess` policy attached:
++
+[NOTE]
+====
+You can find the recommended policy in each project's GitHub repository, for example https://github.com/aws-controllers-k8s/s3-controller/blob/main/config/iam/recommended-policy-arn.
```

Comment:
For the note, do you want to show the URL?

Reply:
I believe the intention was to show the URL (this was in the original content). I'm glad to adjust if we think it would be better.